dislib/data/classes.py

```diff
@@ -58,6 +58,8 @@ def subset_size(self, index):
         if self._sizes[index] is None:
             size = compss_wait_on(_subset_size(self._subsets[index]))
             self._sizes[index] = size
+        elif type(self._sizes[index]) == Future:
```

Comment:
I ain't a big fan of comparing with the Future (should be pycompss internal use only). Can't you just do the compss_wait_on without checking if it's a future? (it will just do nothing when it's not a future)

Reply:
I changed it to compare to int instead of Future

Reply:
What is the cost of running compss_wait_on on a regular object?

Reply:
I've talked to sergio and he said the cost depends on the number of objects in the binding. I believe it is best to leave as it is even if the overhead is small.